packages/rancher-monitoring/generated-changes/patch/values.yaml.patch

```diff
@@ -229,41 +245,33 @@
 +        key: node-role.kubernetes.io/master
 +        operator: "Equal"
 +
-+## Component scraping nginx-ingress-controller
-+##
-+ingressNginx:
++rke2IngressNginx:
 +  enabled: false
-+
 +  ## The namespace to search for your nginx-ingress-controller
-+  ##
-+  namespace: ingress-nginx
-+  
-+  service:
-+    port: 9913
-+    targetPort: 10254
-+    # selector:
-+    #   app: ingress-nginx
-+  serviceMonitor:
-+    ## Scrape interval. If not set, the Prometheus default scrape interval is used.
-+    ##
-+    interval: ""
-+
-+    ## 	metric relabel configs to apply to samples before ingestion.
-+    ##
-+    metricRelabelings: []
-+    # - action: keep
-+    #   regex: 'kube_(daemonset|deployment|pod|namespace|node|statefulset).+'
-+    #   sourceLabels: [__name__]
-+
-+    # 	relabel configs to apply to samples before ingestion.
-+    ##
-+    relabelings: []
-+    # - sourceLabels: [__meta_kubernetes_pod_node_name]
-+    #   separator: ;
-+    #   regex: ^(.*)$
-+    #   targetLabel: nodename
-+    #   replacement: $1
-+    #   action: replace
++  metricsPort: 10254
++  component: ingress-nginx
++  clients:
++    port: 10015
++    useLocalhost: true
++    deployment:
++      replicas: 1
++      affinity:
++        podAffinity:
++          requiredDuringSchedulingIgnoredDuringExecution:
++            - labelSelector:
++                matchExpressions:
++                  - key: "app.kubernetes.io/component"
++                    operator: "In"
++                    values:
++                      - "controller"
++              topologyKey: "kubernetes.io/hostname"
++              namespaces:
++                - "kube-system"
```

Comment:
Author's comment:
1. The nginx-controller is a Deployment with 1 pod in RKE2 cluster which version <= 1.20, and swicthes to Daemonset as of 1.21
2. `podAffinity` makes the pod to be assigned to the node where the nginx-controller's pod runs.